railties/lib/rails/generators/rails/resource_route/resource_route_generator.rb

```diff
@@ -21,7 +25,9 @@ def add_resource_route
         end
 
         # inserts the primary resource
-        write("resources :#{file_name.pluralize}", route_length + 1)
+        resource_string = "resources :#{file_name.pluralize}"
+        resource_string << ", except: [:new, :edit]" if options.api?
```

Comment:
In order to confirm we're covering all the cases, let me break it down:

- `resources` behavor
  - In a regular app: it generates 7 routes
  - In an API-only app: it generates 5 routes (excluding `new` and `edit` implicitily)
  - In an engine, it always generates 7 routes (without taking into consideration the app config `api_only` value)

- scaffold generators with the `--api` option
  - Running the generator in a regular app: it adds a `resources` line to routes.rb including `except: [:new, :edit]`
  - Running the generator in an API-only app: it adds a `resources` line to routes.rb without `except: [:new, :edit]`
  - Running the generator in an engine, it adds a `resources` line to routes.rb including `except: [:new, :edit]`

Is it the plan we're following?